File: typescript/rna-transcription/rna-transcription.ts
export default class Transcriptor {
  toRna(dna: string): string {
    return Array.from(dna, this.dnaToRna).join('')
  }

  private dnaToRna(nucleotide: string): string {
    switch (nucleotide) {
      case 'G': return 'C'
      case 'C': return 'G'
      case 'T': return 'A'
      case 'A': return 'U'
      default: throw new Error('Invalid input DNA.')
    }
  }
}

File: julia/rna-transcription/rna-transcription.jl
MAPPING = Dict('G' => 'C', 'C' => 'G', 'T' => 'A', 'A' => 'U')

function to_rna(dna)
    join(haskey(MAPPING, n) ? MAPPING[n] : error("Invalid") for n in dna)
end

